internal/cmd/commands/database/flags_test.go

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package database
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestInitFlags_SkipInitialLoginRoleCreation(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name           string
+		flags          *initFlags
+		expectedSkip   bool
+		expectedReason string
+	}{
+		{
+			name:           "SkipInitialLoginRoleCreation is true",
+			flags:          &initFlags{flagSkipInitialLoginRoleCreation: true},
+			expectedSkip:   true,
+			expectedReason: reasonSkipInitialLoginRoleCreation,
+		},
+		{
+			name:           "SkipInitialLoginRoleCreation is false",
+			flags:          &initFlags{flagSkipInitialLoginRoleCreation: false},
+			expectedSkip:   false,
+			expectedReason: "",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			skip, reason := tt.flags.SkipInitialLoginRoleCreation()
```

Comment:
If you want these test cases to run in parallel you'll need to add it here too
```suggestion
		t.Run(tt.name, func(t *testing.T) {
			t.Parallel()
			skip, reason := tt.flags.SkipInitialLoginRoleCreation()
```
Same for the rest of these obviously.